services/API-service/src/api/country/country.service.spec.ts

```diff
@@ -0,0 +1,293 @@
+  let countryRepository: Repository<CountryEntity>;
+  let disasterRepository: Repository<DisasterEntity>;
+  let countryDisasterSettingsRepository: Repository<CountryDisasterSettingsEntity>;
+  let leadTimeRepository: Repository<LeadTimeEntity>;
+  let notificationInfoRepository: Repository<NotificationInfoEntity>;
+
+  const relations = [
+    'countryDisasterSettings',
+    'countryDisasterSettings.activeLeadTimes',
+    'disasterTypes',
+    'notificationInfo',
+  ];
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        CountryService,
+        {
+          provide: getRepositoryToken(CountryEntity),
+          useClass: Repository,
+        },
+        {
+          provide: getRepositoryToken(DisasterEntity),
+          useClass: Repository,
+        },
+        {
+          provide: getRepositoryToken(CountryDisasterSettingsEntity),
+          useClass: Repository,
+        },
+        {
+          provide: getRepositoryToken(LeadTimeEntity),
+          useClass: Repository,
+        },
+        {
+          provide: getRepositoryToken(NotificationInfoEntity),
+          useClass: Repository,
+        },
+      ],
+    }).compile();
+
+    service = module.get<CountryService>(CountryService);
+    countryRepository = module.get<Repository<CountryEntity>>(
+      getRepositoryToken(CountryEntity),
+    );
+    disasterRepository = module.get<Repository<DisasterEntity>>(
+      getRepositoryToken(DisasterEntity),
+    );
+    countryDisasterSettingsRepository = module.get<
+      Repository<CountryDisasterSettingsEntity>
+    >(getRepositoryToken(CountryDisasterSettingsEntity));
+    leadTimeRepository = module.get<Repository<LeadTimeEntity>>(
+      getRepositoryToken(LeadTimeEntity),
+    );
+    notificationInfoRepository = module.get<Repository<NotificationInfoEntity>>(
+      getRepositoryToken(NotificationInfoEntity),
+    );
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('getCountries', () => {
+    it('should return an array of countries', async () => {
+      const result = [new CountryEntity()];
+      jest.spyOn(countryRepository, 'find').mockResolvedValue(result);
+
+      expect(await service.getCountries()).toBe(result);
+      expect(countryRepository.find).toHaveBeenCalledWith({ relations });
+    });
+
+    it('should return an array of countries with specific country codes', async () => {
+      const countryCodes = countries
+        .map((country) => country.countryCodeISO3)
+        .join(',');
+      const result = [new CountryEntity()];
+      jest.spyOn(countryRepository, 'find').mockResolvedValue(result);
+
+      expect(await service.getCountries(countryCodes)).toBe(result);
+      expect(countryRepository.find).toHaveBeenCalledWith({
+        where: { countryCodeISO3: In(countryCodes.split(',')) },
+        relations,
+      });
+    });
+  });
+
+  describe('findOne', () => {
```

Comment:
I see this findOne method is actually only public at the moment for the sake of the unit test, which is a mistake. It should have been private, and therefore not be unit tested. It is only used by the method addOrUpdateNotificationInfo , so we can easily delete it. If you want to preserve the 2 different scenarios you now have in findOne, you could move those to the addOrUpdateNotificationInfo  test?

Reply:
> for the sake of the unit test

Did we have unit tests before? I couldn't find any. Anyhow, good catch, I've removed it as you suggested and removed its test.

Reply:
Good. I did not mean there were unit tests before, I meant that it was incorrectly public before, which allowed us to write a unit test on it inadvertently. Anyway, resolved now!